src/ansys_sphinx_theme/__init__.py

```diff
@@ -107,6 +109,113 @@ def setup_default_html_theme_options(app):
+    The link will point to the corresponding file on the main branch.
+
+    Parameters
+    ----------
+    app : Sphinx
+        The Sphinx application instance for rendering the documentation.
+
+    pagename : str
+        The name of the current page.
+
+    templatename : str
+        The name of the template being used.
+
+    context : dict
+        The context dictionary for the page.
+
+    doctree : document
+        The document tree for the page.
+
+    References
+    ----------
+    .. [1] GitHub pull request: https://github.com/pyvista/pyvista/pull/4113
+       (Author: akaszynski <https://github.com/akaszynski>)
+    """
+
+    def fix_edit_link_button(link: str) -> str:
+        """Transform "edit on GitHub" links to the correct URL.
+
+        This function will create the correct URL for the "edit this page" link.
+
+        Parameters
+        ----------
+        link : str
+            The link to the GitHub edit interface.
+
+        Returns
+        -------
+        str
+            The link to the corresponding file on the main branch.
+        """
+        github_user = context.get("github_user", "")
+        github_repo = context.get("github_repo", "")
+        version = getattr(app.builder.env.config, "version", None)
+        if "dev" in version:
+            kind = "main"
+        else:  # pragma: no cover
+            kind = "release/%s" % (".".join(library_version.split(".")[:2]))
+
+        if pagename.startswith("examples") and "index" not in pagename:
+            return f"http://github.com/{github_user}/{github_repo}/edit/{kind}/{pagename}.py"
+        elif "_autosummary" in pagename:
+            domain_keys = {
+                "py": ["module", "fullname"],
+                "c": ["names"],
+                "cpp": ["names"],
+                "js": ["object", "fullname"],
+            }
+
+            for objnode in list(doctree.findall(addnodes.desc)):
+                domain = objnode.get("domain")
+                uris: set[str] = set()
+                for signode in objnode:
+                    if not isinstance(signode, addnodes.desc_signature):
+                        continue
+
+                    # Convert signode to a specified format
+                    info = {}
+                    for key in domain_keys.get(domain, []):
+                        value = signode.get(key)
+                        if not value:
+                            value = ""
+                        info[key] = value
+                    if not info:
+                        continue
+                    # This is an API example
+                    return sphinx_linkcode_resolve(
+                        domain=domain,
+                        info=info,
+                        library=f"{github_user}/{github_repo}",
+                        version=version,
+                        edit=True,
+                    )
+
+        elif "autoapi" in pagename:
+            domain = objnode.get("domain")
```

Comment:
Where is this `objnode` variable first assigned?

The one above is from the `for` loop but does not apply here.

I might be missing something.